blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/README.md

```diff
@@ -0,0 +1,21 @@
+# NOTE
+
+## Datastax driver and Netty Compatibility
+
+The Datastax driver uses Netty for communicating with the Cassandra nodes. Blueflood also uses Netty for the Http layer (reading/writing requests/responses). 
+
+In the past, to resolve conflict in versions, we chose to use Datastax driver with classifier 'shaded'. This essentially shades the Netty classes used by Datastax driver to have a different package name. Datastax doc clearly indicated that if the 'shaded' jar is used, then Netty can not be configured to use native Epoll, which has some performance gains, according to documentation.
+
+Therefore, in order for us leverage native Epoll, we deliberately choose the non-shaded jar of Datastax driver. But we must do so with care. We have to ensure that the Datastax driver for Cassandra is compatible with ours, i.e.: has the same major.minor version. This is the compatiblity matrix:
+
+Cassandra Driver | Built w/ Netty | Blueflood Netty
+Version          | Version        | Version
+-----------------|----------------|----------------
+3.1.2            | 4.0.37.FINAL   | 4.0.40.FINAL
+3.2.0            | 4.0.44.FINAL   | 4.0.44.FINAL
```

Comment:
I made a mistake of not putting this on a branch in my fork. So let me do that, then close the PR. If we decide we want to use native epoll, the changes should be there on a branch. We may not be able to see great improvement from using native epoll until we upgrade to later version of Linux kernel. We are using Ubuntu 13.x, which is using Linux kernel 3.x. There are a lot more performance improvements and troubleshooting tools in Linux kernel 4.x, which is what Ubuntu 16.x (and maybe 14.x) is built with. 

You know what I think of shaded jars. We have 3 versions of the netty jar in JVM. One for blueflood, one for datastax, and one for elasticsearch. That's insane, to me. :-)